lib/fifo.ts

```diff
@@ -1,12 +1,19 @@
 import StaticFIFO from "./static.ts";
 
+export const END = Symbol("Stream ended.");
+export const ERROR = Symbol("Stream errored.");
```

Comment:
Hmm, I'm not quite sure about these. I guess for convenience these are nice to have in some use cases, but I think it makes more sense to leave it out of the implementation and up to usage. I.e., users could create symbols like these themselves and check for them while iterating.